drivers/imu/iio_adis.c

```diff
@@ -1089,7 +1095,11 @@ static int adis_iio_trigger_push_single_sample(struct adis_iio_dev *iio_adis,
 
 	adis = iio_adis->adis_dev;
 
-	ret = adis_read_burst_data(adis, sizeof(buff), buff, iio_adis->burst_size);
+	ret = adis_read_burst_data(adis, sizeof(buff), buff, iio_adis->burst_size, pop,
+				   burst_request);
+	if (ret == -EAGAIN)
```

Comment:
an inline comment might be useful to explain why the treat an error code as success.